maestro-orchestra/src/test/java/maestro/orchestra/yaml/YamlCommandReaderTest.kt

```diff
@@ -1,202 +1,264 @@
 package maestro.orchestra.yaml
 
 import com.google.common.truth.Truth.assertThat
-import com.google.common.truth.Truth.assertWithMessage
 import maestro.orchestra.ApplyConfigurationCommand
 import maestro.orchestra.BackPressCommand
 import maestro.orchestra.Command
-import maestro.orchestra.error.InvalidInitFlowFile
 import maestro.orchestra.LaunchAppCommand
 import maestro.orchestra.MaestroCommand
 import maestro.orchestra.MaestroConfig
 import maestro.orchestra.MaestroInitFlow
 import maestro.orchestra.ScrollCommand
+import maestro.orchestra.error.InvalidInitFlowFile
 import maestro.orchestra.error.SyntaxError
-import org.junit.Rule
-import org.junit.Test
-import org.junit.rules.TestName
+import maestro.orchestra.yaml.junit.YamlCommandsExtension
+import maestro.orchestra.yaml.junit.YamlExceptionExtension
+import maestro.orchestra.yaml.junit.YamlFile
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
 import java.nio.file.FileSystems
 import java.nio.file.Paths
 
-@Suppress("TestFunctionName")
+@Suppress("JUnitMalformedDeclaration")
+@ExtendWith(YamlCommandsExtension::class, YamlExceptionExtension::class)
 internal class YamlCommandReaderTest {
 
-    @JvmField
-    @Rule
-    val name: TestName = TestName()
-
     @Test
-    fun T001_empty() = expectException<SyntaxError> { e ->
+    fun empty(
+        @YamlFile("001_empty.yaml") e: SyntaxError,
```

Comment:
Should we call this `@Dsl` or `@FlowFile` instead of `@YamlFile`?

Reply:
I think `@YamlFile` is great, it is explicit